src/index.ts

```diff
@@ -371,6 +377,178 @@ export const createWapeSaleInstance = (
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const status: GenSaleStatus = await genSaleActions.getSaleStatus(contract);
+      return status;
+    },
+    getMintlist: async (): Promise<Mintlist> => {
+      const whitelist = await getMintlist();
+      return whitelist;
+    },
+    getMintlistedUserClaim: async (address: string): Promise<Claim> => {
+      const mintlist = await getMintlist();
+      const userClaim: Maybe<Claim> = mintlist.claims[address];
+      if (!userClaim) {
+        throw Error(
+          `No claim could be found for user ${address} because they are not on the mintlist`
+        );
+      }
+      return userClaim;
+    },
+    getTotalForSale: async (): Promise<number> => {
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const total = await contract.amountForSale();
+
+      return total.toNumber();
+    },
+    getNumberOfDomainsSold: async (): Promise<number> => {
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const domainsSold = await contract.domainsSold();
+      return domainsSold.toNumber();
+    },
+    getBlockNumber: async (): Promise<number> => {
+      const blockNum = await config.web3Provider.getBlockNumber();
+      return blockNum;
+    },
+    getEthBalance: async (address: string): Promise<string> => {
+      const balance = await config.web3Provider.getBalance(address);
+      return ethers.utils.formatEther(balance);
+    },
+    isUserOnMintlist: async (address: string): Promise<boolean> => {
+      const mintlist = await getMintlist();
+      const isOnWhitelist = mintlist.claims[address] ? true : false;
+      return isOnWhitelist;
+    },
+    getDomainsPurchasedByAccount: async (address: string): Promise<number> => {
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const domains = await contract.domainsPurchasedByAccount(address);
+      return domains.toNumber();
+    },
+    purchaseDomains: async (
+      count: ethers.BigNumber,
+      signer: ethers.Signer
+    ): Promise<ethers.ContractTransaction> => {
+      const contract = await getGenSaleContract(
+        signer,
+        config.contractAddress
+      );
+
+      const mintlist = await getMintlist();
+
+      const tx = await genSaleActions.purchaseDomains(
+        count,
+        signer,
+        contract,
+        mintlist
+      );
+      return tx;
+    },
+    setPauseStatus: async (
+      pauseStatus: boolean,
+      signer: ethers.Signer
+    ): Promise<ethers.ContractTransaction> => {
+      const contract = await getGenSaleContract(
+        signer,
+        config.contractAddress
+      );
+      const tx = await genSaleActions.setPauseStatus(
```

Comment:
we can just `return` in cases like these without caching into a var. it will be awaited on the side of the caller.

Reply:
@JamesEarle likes to assign responses like these to a variable for easier debugging. I do it this way because Ive been asked to in the past, personally it doesn't matter to me.

Reply:
haha makes sense for debugging. in smaller modules like these it's really up to the dev. but keeping memory in mind, especially in big objects like this might be beneficial.
during debugging we can always just assign it when needed to get the value here.

Reply:
@Whytecrowe Yeah will is right, I prefer to assign our return values to variables for ease of debugging in the future 😁